lib/wasi/src/syscalls/wasix/futex_wait.rs

```diff
@@ -3,56 +3,95 @@ use std::task::Waker;
 use super::*;
 use crate::syscalls::*;
 
-struct FutexPoller<'a, M>
-where
-    M: MemorySize,
-{
-    env: &'a WasiEnv,
-    view: MemoryView<'a>,
+#[derive(Clone)]
+struct FutexPoller {
+    state: Arc<WasiState>,
+    woken: Arc<Mutex<bool>>,
```

Comment:
AtomicBool seems better suited here?

[std](https://doc.rust-lang.org/std/index.html)::[sync](https://doc.rust-lang.org/std/sync/index.html)::[atomic](https://doc.rust-lang.org/std/sync/atomic/index.html)::[AtomicBool](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicBool.html#)

Reply:
i managed to delete it completely instead